### PR TITLE
fixes for ENABLE_IMPLICIT_CONVERSIONS=Off

### PR DESCRIPTION
**Pull request by dtaller**
In the future, ale3d plans to use chai with ENABLE_IMPLICIT_CONVERSIONS off. Having this option on (as is the case by default) has been the source of a bunch of gpu-related bugs in ale3d. When an int* is converted to a managed array implicitly without passing an array length, errors occur as data is copied to/from the gpu.

### CODE REVIEW

File: src/chai/managed_ptr.hpp [hunk 1]
```diff
@@ -664,7 +664,7 @@ namespace chai {
       ///
       template <typename T>
       CHAI_HOST_DEVICE T* getRawPointers(ManagedArray<T> arg) {
-         return (T*) arg;
+         return arg.getActivePointer();
```
Comment (adayton1) [inline src/chai/managed_ptr.hpp:667]: This is incorrect. `getActivePointer()` does not move data to the correct space. I have a pull request in that adds the "data" method that does the same thing as the cast. Once that is in, you can safely use that here. `return arg.data();`

File: tests/unit/array_manager_unit_tests.cpp [hunk 2]
```diff
@@ -18,6 +18,8 @@ TEST(ArrayManager, Constructor)
 
 #ifndef CHAI_DISABLE_RM
 
+#if defined(CHAI_ENABLE_IMPLICIT_CONVERSIONS)
```
Comment (adayton1) [inline tests/unit/array_manager_unit_tests.cpp:21]: This test shouldn't be relying on implicit conversions. Once my other pull request goes in, use the data method.

File: src/chai/managed_ptr.hpp (hunk 1 above)
Comment (dtaller) [reply src/chai/managed_ptr.hpp]: Hi @adayton1 . Should be fixed

File: tests/unit/array_manager_unit_tests.cpp (hunk 2 above)
Comment (dtaller) [reply tests/unit/array_manager_unit_tests.cpp]: Done, fixed

**Pull request merged**